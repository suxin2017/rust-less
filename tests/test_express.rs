#[cfg(feature = "serde")]
use lightningcss::printer::PrinterOptions;
#[cfg(feature = "serde")]
use lightningcss::stylesheet::{ParserOptions, StyleSheet};
#[cfg(feature = "serde")]
use lightningcss::targets::{Browsers, Features, Targets};

#[cfg(feature = "serde")]
macro_rules! test_ast {
  ($code:expr) => {
    let stylesheet = StyleSheet::parse($code, ParserOptions::default()).unwrap();
    let targets = Targets {
      browsers: Some(Browsers {
        chrome: Some(95 << 16),
        ..Browsers::default()
      }),
      include: Features::all(),
      exclude: Features::empty(),
    };
    let res = stylesheet
      .to_css(PrinterOptions {
        targets,
        ..PrinterOptions::default()
      })
      .unwrap();

    insta::assert_snapshot!(res.code);
  };
}

#[cfg(feature = "serde")]
#[test]
fn test() {
  test_ast!(
    r#"
a{
width:1;
width: 1+ 2;
width: 1+ 2 * 3;
width: 1+ 2 * 3 +4;
width: 1+ 3 /2 +3;
width: 1+ 2 * (1 + 3);
wdith: 2/(9)+6*9;
width: (8)/(1)/(2)/(2);
width: 7-4+7-4;
width: (7)+(7)+3+(1);
width: 5*(5)+8+(2);
width:1*(9)-(5)*6;
width:(2)+(4)/(5)+(2);
width:1/7/1/2;
width:(6)-(7)*6/(8);
width:(1)-9/3+6;
}
  "#
  );
}
