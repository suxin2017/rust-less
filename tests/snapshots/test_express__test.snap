---
source: tests/test_express.rs
expression: res.code
---
a {
  width: 1px;
  width: 3;
  width: 7;
  width: 11;
  width: 5.5;
  width: 9;
  wdith: 54.22222;
  width: 2;
  width: 6;
  width: 18;
  width: 35;
  width: -21;
  width: 4.8;
  width: 0.071428575;
  width: 0.75;
  width: 4;
}

